File: WindowDisplayWebservice/app/src/main/java/com/orbits/queuingsystem/AppConstantsFlags.java
package com.orbits.queuingsystem;

class AppConstantsFlags {
    static final String key_pref_show_ad = "SHOW_ADS";
    static final String key_pref_show_scroll = "SHOW_SCROLL";
    static final String key_pref_show_logo = "SHOW_LOGO";
    static final String key_pref_restart = "RESTART_APP";
    static final String key_pref_time_to_restart = "TIME_TO_RESTART";
    static final String key_pref_show_counter = "SHOW_COUNTER";
    static final String KEY_DIGIT_MAP_ARB_FONT_HALF_SCREEN = "Arabic HalfScr FontSize";
    static final String KEY_DIGIT_MAP_ENG_FONT_HALF_SCREEN = "English HalfScr FontSize";
    static final String KEY_DIGIT_MAP_ARB_FONT_FULL_SCREEN = "Arabic FullScr FontSize";
    static final String KEY_DIGIT_MAP_ENG_FONT_FULL_SCREEN = "English FullScr FontSize";
    static final String KEY_DIGIT_MAP_BG_COLOR = "Background Color";
    static final String KEY_DIGIT_MAP_TOKEN_COLOR = "Token Color";
    static final String KEY_DIGIT_MAP_TOKEN_BLINK_COLOR = "Token Blink Color";
    static final String KEY_DIGIT_MAP_TOKEN_PADDING_TOP = "Token Padding Top";
    static final String KEY_DIGIT_MAP_TOKEN_PADDING_BOTTOM = "Token Padding Bottom";
    static final String KEY_DIGIT_MAP_COUNTER_PADDING_BOTTOM = "Counter Padding Bottom";
    static final String KEY_DIGIT_MAP_COUNTER_PADDING_TOP = "Counter Padding Top";
    static final String KEY_DIGIT_MAP_COUNTER_COLOR = "Counter Color";
    static final String KEY_DIGIT_MAP_COUNTER_BG_COLOR = "Counter Background Color";
    static final String KEY_DIGIT_MAP_ENG_COUNTER_FONT_FULL_SCREEN = "Counter English FullScr FontSize";
    static final String KEY_DIGIT_MAP_ENG_COUNTER_FONT_HALF_SCREEN = "Counter English HalfScr FontSize";
    static final String KEY_DIGIT_MAP_ARB_COUNTER_FONT_HALF_SCREEN = "Counter Arabic HalfScr FontSize";
    static final String KEY_DIGIT_MAP_ARB_COUNTER_FONT_FULL_SCREEN = "Counter Arabic FullScr FontSize";
    static final String KEY_DIGIT_MAP_SERVICE_NAME_TEXT_COLOR = "Service Text Color";
    static final String KEY_DIGIT_MAP_SERVICE_NAME_FONT_SIZE = "Service FontSize";
    static final String KEY_DIGIT_MAP_BRANCH_NAME_TEXT_COLOR = "Branch Text Color";
    static final String KEY_DIGIT_MAP_BRANCH_NAME_BG_COLOR = "Branch Background Color";
    static final String KEY_DIGIT_MAP_BRANCH_NAME_FONT_SIZE = "Branch FontSize";
    static final String KEY_DIGIT_MAP_TIME_FONT_SIZE = "Time FontSize";
    static final String KEY_DIGIT_MAP_TIME_TEXT_COLOR = "Time Text Color";



    static final int SOCKET_HANDLER_DELAY = 1000;
    static final int SOCKET_BUFFER_SIZE = 512;
    static final int SOCKET_TIMEOUT_INTERVAL = 3000;
    static final int DEVICE_TYPE_ANDROID_WINDOW_DISPLAY = 3;
    static final int COMMAND_CODE_IDENTIFY_SELF = 58;
    static final int STATUS_BYTE = 128;
    static final String KEY_DIGIT_MAP_AUDIO_FILE_EXTENSION = "Audio-File extension";
    static final String KEY_CC = "Customer_Code";
    static final String KEY_IN = "Invoice_No";
    static final String KEY_BC = "Branch_Code";
    static final String KEY_EMAIL = "Email_Id";
    static final String KEY_CONTACT_NO = "Phone_No";
    static final String KEY_STAFF_ID = "STAFF_ID";

    //SharedPref
    static final String KEY_PASSWORD_STAFF = "PASSWORD_STAFF";
    public static String key_pref_service_name = "SERVICE_NAME";
    public static String key_pref_branch_name = "BRANCH_NAME";
    public static String key_pref_user_name = "USER_NAME";
    public static String key_pref_show_time= "TIME";
    // nirmal 17 june 2014 -- common class to declare global constants and flags
    static String SHOW_ADVERTIZEMENT = "";
    static boolean SHOW_SCROLL = true;
    static boolean SHOW_LOGO = true;
    static boolean SHOW_SERVICE_NAME = true;
    static boolean SHOW_BRANCH_NAME = true;
    static boolean SHOW_TIME = true;
    static boolean ANDROID_SERVER;
    static boolean SEND_PROTOCOL_TO_SERVER = false;
    static boolean READ_PROTOCOL_FROM_SERVER = false;
}

File: app/src/main/java/com/orbits/queuingsystem/AppConstantsFlags.kt
package com.orbits.queuingsystem

internal object AppConstantsFlags {
    const val key_pref_show_ad: String = "SHOW_ADS"
    const val key_pref_show_scroll: String = "SHOW_SCROLL"
    const val key_pref_show_logo: String = "SHOW_LOGO"
    const val key_pref_restart: String = "RESTART_APP"
    const val key_pref_time_to_restart: String = "TIME_TO_RESTART"
    const val key_pref_show_counter: String = "SHOW_COUNTER"
    const val KEY_DIGIT_MAP_ARB_FONT_HALF_SCREEN: String = "Arabic HalfScr FontSize"
    const val KEY_DIGIT_MAP_ENG_FONT_HALF_SCREEN: String = "English HalfScr FontSize"
    const val KEY_DIGIT_MAP_ARB_FONT_FULL_SCREEN: String = "Arabic FullScr FontSize"
    const val KEY_DIGIT_MAP_ENG_FONT_FULL_SCREEN: String = "English FullScr FontSize"
    const val KEY_DIGIT_MAP_BG_COLOR: String = "Background Color"
    const val KEY_DIGIT_MAP_TOKEN_COLOR: String = "Token Color"
    const val KEY_DIGIT_MAP_TOKEN_BLINK_COLOR: String = "Token Blink Color"
    const val KEY_DIGIT_MAP_TOKEN_PADDING_TOP: String = "Token Padding Top"
    const val KEY_DIGIT_MAP_TOKEN_PADDING_BOTTOM: String = "Token Padding Bottom"
    const val KEY_DIGIT_MAP_COUNTER_PADDING_BOTTOM: String = "Counter Padding Bottom"
    const val KEY_DIGIT_MAP_COUNTER_PADDING_TOP: String = "Counter Padding Top"
    const val KEY_DIGIT_MAP_COUNTER_COLOR: String = "Counter Color"
    const val KEY_DIGIT_MAP_COUNTER_BG_COLOR: String = "Counter Background Color"
    const val KEY_DIGIT_MAP_ENG_COUNTER_FONT_FULL_SCREEN: String =
        "Counter English FullScr FontSize"
    const val KEY_DIGIT_MAP_ENG_COUNTER_FONT_HALF_SCREEN: String =
        "Counter English HalfScr FontSize"
    const val KEY_DIGIT_MAP_ARB_COUNTER_FONT_HALF_SCREEN: String = "Counter Arabic HalfScr FontSize"
    const val KEY_DIGIT_MAP_ARB_COUNTER_FONT_FULL_SCREEN: String = "Counter Arabic FullScr FontSize"
    const val KEY_DIGIT_MAP_SERVICE_NAME_TEXT_COLOR: String = "Service Text Color"
    const val KEY_DIGIT_MAP_SERVICE_NAME_FONT_SIZE: String = "Service FontSize"
    const val KEY_DIGIT_MAP_BRANCH_NAME_TEXT_COLOR: String = "Branch Text Color"
    const val KEY_DIGIT_MAP_BRANCH_NAME_BG_COLOR: String = "Branch Background Color"
    const val KEY_DIGIT_MAP_BRANCH_NAME_FONT_SIZE: String = "Branch FontSize"
    const val KEY_DIGIT_MAP_TIME_FONT_SIZE: String = "Time FontSize"
    const val KEY_DIGIT_MAP_TIME_TEXT_COLOR: String = "Time Text Color"


    const val SOCKET_HANDLER_DELAY: Int = 1000
    const val SOCKET_BUFFER_SIZE: Int = 512
    const val SOCKET_TIMEOUT_INTERVAL: Int = 3000
    const val DEVICE_TYPE_ANDROID_WINDOW_DISPLAY: Int = 3
    const val COMMAND_CODE_IDENTIFY_SELF: Int = 58
    const val STATUS_BYTE: Int = 128
    const val KEY_DIGIT_MAP_AUDIO_FILE_EXTENSION: String = "Audio-File extension"
    const val KEY_CC: String = "Customer_Code"
    const val KEY_IN: String = "Invoice_No"
    const val KEY_BC: String = "Branch_Code"
    const val KEY_EMAIL: String = "Email_Id"
    const val KEY_CONTACT_NO: String = "Phone_No"
    const val KEY_STAFF_ID: String = "STAFF_ID"

    //SharedPref
    const val KEY_PASSWORD_STAFF: String = "PASSWORD_STAFF"
    var key_pref_service_name: String = "SERVICE_NAME"
    var key_pref_service_top: String = "SERVICE_TOP"
    var key_pref_none_top: String = "NONE_TOP"
    var key_pref_user_top: String = "USER_TOP"
    var key_pref_branch_top: String = "BRANCH_TOP"
    var key_pref_branch_name: String = "BRANCH_NAME"
    var key_pref_none: String = "NONE"
    var key_pref_show_time: String = "TIME"
    var key_pref_user_name: String = "USER_NAME"

    // nirmal 17 june 2014 -- common class to declare global constants and flags
    var SHOW_ADVERTIZEMENT: String = ""
    var SHOW_SCROLL: Boolean = true
    var SHOW_LOGO: Boolean = true
    var SHOW_SERVICE_NAME: Boolean = true
    var SHOW_SERVICE_NAME_TOP: Boolean = true
    var SHOW_USER_NAME_TOP: Boolean = true
    var SHOW_BRANCH_NAME_TOP: Boolean = true
    var SHOW_NONE_TOP: Boolean = true
    var SHOW_NONE: Boolean = true
    var SHOW_BRANCH_NAME: Boolean = true
    var SHOW_USER_NAME: Boolean = true
    var SHOW_TIME: Boolean = true
    var ANDROID_SERVER: Boolean = false
    var SEND_PROTOCOL_TO_SERVER: Boolean = false
    var READ_PROTOCOL_FROM_SERVER: Boolean = false
}
